samples/gbmc_client.py

```diff
@@ -0,0 +1,182 @@
+#
+# Copyright (C) 2013-2019 The ESPResSo project
+#
+# This file is part of ESPResSo.
+#
+# ESPResSo is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ESPResSo is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Client part of the Gibbs ensamble simulation. This script handles the simulation boxes and communicates the energies to the host. The Monte-Carlo part of the simulation is done by the `gbmc_socket.py` script.
+"""
+
+import espressomd
+import socket
+import numpy as np
+import pickle
+import struct
+import random
+import argparse
+
+seed = 1
+init_box_l = 7.55
+particle_radius = 0.5
+particle_number = 128
+
+lj_epsilon = 1.0
+lj_sigma = 2.0 * particle_radius
+lj_cutoff = 3.0
+lj_shift = 0.0
+
+HOST = 'localhost'
+PORT = 31415
+number_of_clients = 2
+
+# Message identifiers
+MSG_START = 0
+MSG_END = 1 
+MSG_MOVE_PART = 2
+MSG_MOVE_PART_REVERT = 21
+MSG_CHANGE_VOLUME = 3
+MSG_EXCHANGE_PART_ADD = 4
+MSG_EXCHANGE_PART_ADD_REVERT = 41
+MSG_EXCHANGE_PART_REMOVE = 5
+MSG_EXCHANGE_PART_REMOVE_REVERT = 51
+MSG_ENEGRY = 6
+
+# parse arguments
+parser = argparse.ArgumentParser()
+parser.add_argument('-n', '--number-particles', type=int, nargs=1)
+parser.add_argument('-lj', '--lj-params', type=float, nargs=4)
+parser.add_argument('-s', '--seed', type=int, nargs=1)
+parser.add_argument('-bl', '--box-length', type=float, nargs=1)
+
+args = parser.parse_args()
+
+if(args.number_particles):
+    particle_number = args.number_particles[0]
+if(args.lj_params):
+    lj_epsilon = args.lj_params[0]
+    lj_sigma = args.lj_params[1]
+    lj_cutoff = args.lj_params[2]
+    lj_shift = args.lj_params[3]
+if(args.seed):
+    seed = args.seed[0]
+if(args.box_length):
+    init_box_l = args.box_length[0]
+
+
+def move_particle():
```

Comment:
Maybe we could use MD steps to propagate the system instead of using MC moves. This is needed for more complex (e.g. polymeric) systems.
You will need to use a thermostat for the MD integrator. Additionally, you will need to be carful about not placing the particles too close to each other (otherwise you leave the stability criterion of the MD integrator). On the other hand, maybe this is a task for somebody actually wanting to employ the gibbs ensemble for production use.

Reply:
has this been tried out?